client/src/App.js

```diff
@@ -13,16 +13,24 @@ export default class App extends Component {
   }
 
   addToSavedList = movie => {
+    //console.log(movie.title)
+    if (this.state.savedList.includes(movie)) {
+      alert(`${movie.title} is already in your list!`)
+    } else {
     const savedList = this.state.savedList;
     savedList.push(movie);
     this.setState({ savedList });
+    }
   };
 
   render() {
     return (
       <div>
         <SavedList list={this.state.savedList} />
-        <div>Replace this Div with your Routes</div>
+        <Route exact path="/" component={MovieList} />
+      
+        <Route  path="/movies/:id" render={(props, index) => <Movie key={index} {...props} addToSavedList={this.addToSavedList}/>} />
+        
       </div>
     );
```

Comment:
Line 32 - considering that `render` is not actually mapping through a set of items, you don't actually have to use the `index` parameter and the `key` prop here.